packages/api/src/routes/createSnippet.ts

```diff
@@ -0,0 +1,91 @@
+import type { TRequest } from '@chatsift/rest-utils';
+import { jsonParser, Route, RouteMethod } from '@chatsift/rest-utils';
+import { REST } from '@discordjs/rest';
+import { badRequest } from '@hapi/boom';
+import { PrismaClient } from '@prisma/client';
+import type { BaseValidator, InferType } from '@sapphire/shapeshift';
+import { s } from '@sapphire/shapeshift';
+import type {
+	RESTPostAPIApplicationCommandsResult,
+	RESTPostAPIApplicationGuildCommandsJSONBody,
+} from 'discord-api-types/v9';
+import { ApplicationCommandOptionType, Routes } from 'discord-api-types/v9';
+import type { Middleware, Response } from 'polka';
+import { singleton } from 'tsyringe';
+import { Env } from '../util/env';
+import type { Snippet } from '../util/models';
+import { snowflakeSchema } from '../util/snowflakeSchema';
+
+const schema = s.object({
+	name: s.string.lengthGreaterThan(0).lengthLessThanOrEqual(32),
+	content: s.string.lengthGreaterThan(0).lengthLessThanOrEqual(1_900),
+	createdById: snowflakeSchema,
+}).strict;
+type Body = InferType<typeof schema>;
+
+@singleton()
+export default class extends Route<Snippet, Body> {
+	public info = {
+		method: RouteMethod.put,
+		path: '/modmail/v1/guilds/:guildId/snippets/',
+	} as const;
+
+	public override readonly bodyValidationSchema: BaseValidator<Body> = schema;
+
+	public override middleware: Middleware[] = [jsonParser()];
+
+	public constructor(
+		private readonly prisma: PrismaClient,
+		private readonly discordRest: REST,
+		private readonly env: Env,
+	) {
+		super();
+	}
+
+	public async handle(req: TRequest<Body>, res: Response) {
+		const { guildId } = req.params as { guildId: string };
+
+		const snippetAlreadyExists = await this.prisma.snippet.findFirst({
+			where: {
+				name: req.body.name,
+				guildId,
+			},
+		});
+		if (snippetAlreadyExists) {
+			throw badRequest('A snippet with this name already exists in the guild');
```

Comment:
Prefer you take in `next: NextHandler` in `handle`

```suggestion
			return next(badRequest('A snippet with this name already exists in the guild'));
```

Reply:
Done in 57469db